arcaflow_plugin_coremark_pro/coremark_pro_plugin.py

```diff
@@ -63,17 +133,26 @@ def certify_all(
         "CoreMark-PRO": {},
     }
 
+    # Construct the output object
     for line in ca_return[1].splitlines():
+        line_list = line.split()
         try:
-            line_name = line.split()[0]
+            line_name = line_list[0]
         except IndexError:
-            pass
-        if line_name in ca_results and len(line.split()) > 1:
+            continue
+        if line_name in ca_results:
             ca_results[line_name] = {
-                "MultiCore": line.split()[1],
-                "SingleCore": line.split()[2],
-                "Scaling": line.split()[3],
+                "MultiCore": float(line_list[1]),
+                "SingleCore": float(line_list[2]),
+                "Scaling": float(line_list[3]),
             }
+            # Collect the per-benchmark iterations from the log file
+            if line_name != "CoreMark-PRO":
+                with open(file=run_log_path, encoding="utf-8") as file:
+                    for line in file:
+                        line_list = line.split()
+                        if "median single" in line and line_list[2] == line_name:
+                            ca_results[line_name]["Iterations"] = int(line_list[7])
```

Comment:
I don't know how big the log file is, but re-opening and reading _every_ line of it looking for each line in the `ca_results` seems unsatisfying.  It would be better to move the log file search out of the loop at line 137 and make a single pass over it:
```python
    with open(file=run_log_path, encoding="utf-8") as file:
        for line in file:
            if "median single" not in line or "CoreMark-PRO" in line:
                continue
            line_list = line.split()
            if line_list[2] in ca_results:
                 ca_results[line_list[2]]["Iterations"] = int(line_list[7])
```
This opens the file only once, and reads all the lines only once.  If there is no median or if it's the PRO score, we skip to the next line, rather than creating the split.  Finally, if the benchmark is one that we're looking for, add the iteration count to it using a dict lookup instead of a linear search.